crates/matrix-sdk-base/src/response_processors/state_events.rs

```diff
@@ -317,31 +309,47 @@ where
         .unzip()
 }
 
-/// Check if `m.room.create` isn't creating a loop of rooms.
-pub fn is_create_event_valid(
+/// Check if the `predecessor` in `m.room.create` isn't creating a loop of
+/// rooms.
+///
+/// If it is, we return a clone of the event with the predecessor removed.
+pub fn validate_create_event_predecessor(
     context: &mut Context,
     room_id: &RoomId,
     event: &SyncStateEvent<RoomCreateEventContent>,
     state_store: &BaseStateStore,
-) -> bool {
+) -> Option<SyncStateEvent<RoomCreateEventContent>> {
     let mut already_seen = BTreeSet::new();
     already_seen.insert(room_id.to_owned());
 
-    let Some(mut predecessor_room_id) = event
-        .as_original()
-        .and_then(|event| Some(event.content.predecessor.as_ref()?.room_id.clone()))
+    // Redacted and non-redacted create events use the same content type.
+    let content = match event {
+        SyncStateEvent::Original(event) => &event.content,
+        SyncStateEvent::Redacted(event) => &event.content,
+    };
```

Comment:
I think `m.room.create` cannot be redacted. That's why `as_original()` was called previously. I'm fine with the change (better be safe), but maybe it should be mentioned.

Reply:
It can be redacted, but redaction has no effect in recent room versions. Given that Ruma relies on the detection of `redacted_because` in `unsigned`, not on the format of the content, it is entirely possible to get the `Redacted` variant here.